FrontEnd/src/context/AuthContextProvider.js

```diff
@@ -20,22 +22,23 @@ export function AuthProvider ({ children }) {
           Authorization: `Token ${authToken}`,
         },
       })
-      .then(res => res.data)
-      .catch((error) => {
-        if (error.response && error.response.status === 401) {
-          logout();
-        }
-        console.error('An error occurred while fetching the data.', error);
-      },
-    { revalidateOnFocus: false }
-  )
-);
+        .then(res => res.data)
+        .catch((error) => {
+          if (error.response && error.response.status === 401) {
+            logout();
+          }
+          console.error('An error occurred while fetching the data.', error);
+        },
+          { revalidateOnFocus: false }
+        )
+  );
 
   const login = (authToken) => {
     localStorage.setItem('Token', authToken);
     setAuthToken(authToken);
     localStorage.setItem('isAuth', true);
     axios.defaults.headers.common['Authorization'] = `Token ${authToken}`;
+    setIsStaff(checkIfStaff);
```

Comment:
`checkIfStaff` seems to be an asynchronous function. You can't just put it into setState. You need to await for the result and only then call setState

Reply:
Done